python/triton/runtime/jit.py

```diff
@@ -319,15 +319,16 @@ def _make_launcher(self):
         args_signature = ', '.join(name if dflt == inspect._empty else f'{name} = {dflt}' for name, dflt in zip(self.arg_names, self.arg_defaults))
 
         src = f"""
-def {self.fn.__name__}({args_signature}, grid=None, num_warps=4, num_stages=3, extern_libs=None, stream=None, warmup=False, device=None, device_type=None):
+def {self.fn.__name__}({args_signature}, grid=None, extern_libs=None, stream=None, warmup=False, device=None, device_type=None, **kwargs):
     from ..compiler import compile, CompiledKernel
     sig_key =  {sig_keys},
     constexpr_key = {f'{constexpr_keys},' if len(constexpr_keys) > 0 else ()}
     spec_key = {f'{spec_keys},' if len(spec_keys) > 0 else ()}
+    num_warps = kwargs.get("num_warps")
```

Comment:
How about `kwargs.get("num_warps", 4)` and `kwargs.get("num_stages", 3)`?

Reply:
It may have conflict to the real default value being used later in torch.compile if the user not set the num_waprs and num_stages.

The `key = (version_key, sig_key, constexpr_key, spec_key, None, None, self.debug)`  may refer to the kernel with the default num_warps and num_stages based on the device architecture information.

The side effect is that we may have duplicate kernel cached with different key value. But logically there is no issue and transparent to the user.

Reply:
I don't get it actually, if users do not set `num_stages`, what would be the default value after this PR?

Reply:
The default value of `num_warps` and `num_stages` will be set in the triton.compile, a unified place to chose the default configuration based on the architecture.

Reply:
Yeah, I believe they can be used before getting set in the compile function. Please refer to my comments below.